src/magic/model/MagicAmount.java

```diff
@@ -10,17 +10,35 @@ public int getAmount(final MagicEvent event) {
         return getAmount(event.getSource(), event.getPlayer());
     }
 
+    public int getPositiveAmount(final MagicSource source, final MagicPlayer player) {
+        if (this == MagicAmountFactory.NegXCost) {
+            // Return a negative number for NegXCost
+            return Math.min(getAmount(source, player), 0);
+        } else {
+            return Math.max(getAmount(source, player), 0);
+        }
+    }
+
+    public int getPositiveAmount(final MagicEvent event) {
+        if (this == MagicAmountFactory.NegXCost) {
+            // Return a negative number for NegXCost
+            return Math.min(getAmount(event), 0);
+        } else {
+            return Math.max(getAmount(event), 0);
+        }
+    }
+
     public boolean isConstant() {
         return false;
     }
 
     public static int countEachProduct(final MagicAmount amount, final MagicAmount eachCount, final MagicEvent event) {
```

Comment:
In this method, it does not limit how the result is used, so it should always be getAmount since it could be computing a value to set the life total.